src/ui.rs

```diff
@@ -24,24 +27,37 @@ pub fn draw<B: Backend>(f: &mut Frame<B>, app: &App) {
             .as_ref(),
         )
         .split(f.size());
+
     let titles = app
         .tabs
-        .titles
+        .tabs
         .iter()
-        .map(|t| Spans::from(Span::styled(*t, Style::default().fg(Color::Green))))
+        .map(|tab| {
+            let title = match &tab.algorithm {
+                AlgorithmWrapper::GrahamScan(algorithm) => Spans::from(Span::styled(
+                    algorithm.get_title(),
+                    Style::default().fg(Color::Green),
+                )),
+            };
+            title
+        })
         .collect();
+
     let tabs = Tabs::new(titles)
         .block(Block::default().borders(Borders::ALL).title(app.title))
         .highlight_style(Style::default().fg(Color::Yellow))
         .select(app.tabs.index);
+
     f.render_widget(tabs, chunks[0]);
     draw_header(f, chunks[1]);
 
-    match app.tabs.index {
-        0 => app.algorithm.draw_algorithm(f, chunks[2], app),
-        _ => {}
-    }
-    draw_footer(f, chunks[3], app);
+    let algorithm = match &app.tabs.tabs[app.tabs.index].algorithm {
+        AlgorithmWrapper::GrahamScan(algorithm) => algorithm,
+    };
+
+    GrahamScan::draw_algorithm(algorithm, f, chunks[2], app);
```

Comment:
This should all be abstracted by the `AlgorithmWrapper`. Add a `.draw()` method to the `impl AlgorithmWrapper` block.